src/app/modules/angular-slickgrid/services/grid-odata.service.ts

```diff
@@ -535,77 +537,94 @@ export class GridOdataService implements BackendService {
+            searchValue = searchValue.replace(decimalInputSeparator, '.');
+          }
 
-    // multiple search value (date range)
-    if (Array.isArray(searchValues) && searchValues.length === 2 && searchValues[0] && searchValues[1]) {
-      // date field needs to be UTC and within DateTime function
-      const searchValue1 = this.odataQueryVersionWrapper('dateTime', version, fieldName, parseUtcDate(searchValues[0] as string, true));
-      const searchValue2 = this.odataQueryVersionWrapper('dateTime', version, fieldName, parseUtcDate(searchValues[1] as string, true));
-
-      if (searchValue1 && searchValue2) {
-        if (operator === OperatorType.rangeInclusive) {
-          // example:: (Finish >= DateTime'2019-08-11T00:00:00Z' and Finish <= DateTime'2019-09-12T00:00:00Z')
-          query = `(${fieldName} ge ${searchValue1} and ${fieldName} le ${searchValue2})`;
-        } else if (operator === OperatorType.rangeExclusive) {
-          // example:: (Finish > DateTime'2019-08-11T00:00:00Z' and Finish < DateTime'2019-09-12T00:00:00Z')
-          query = `(${fieldName} gt ${searchValue1} and ${fieldName} lt ${searchValue2})`;
+          // Replace double dots with single dots
+          searchValue = searchValue.replace(/\.\./g, '.');
+          // Remove a trailing dot
+          searchValue = searchValue.replace(/\.+$/g, '');
+          // Predix a leading dot with 0
+          searchValue = searchValue.replace(/^\.+/g, '0.');
+          // Remove any non valid decimal characters from the search string
+          searchValue = searchValue.replace(/(?!^\-)[^\d\.]/g, '');
+
+          // if nothing left, search for 0
+          if (searchValue === '') {
+            searchValue = '0';
+          }
         }
-      }
+        break;
     }
-    return query;
+
+    return searchValue;
   }
 
   /**
-   * Filter by a range of searchTerms (2 searchTerms OR 1 string separated by 2 dots "value1..value2")
+   * Returns all allowed decimal separators.
+   * For example in NL it is common when entering numbers that both ',' and '.' acct as the decimal separator,
+   * while displaying/formatting the ',' is the decimal separator and the '.' the thousand separator.
    */
-  private filterBySearchTermRange(fieldName: string, operator: OperatorType | OperatorString, searchTerms: SearchTerm[]) {
-    let query = '';
-
-    if (Array.isArray(searchTerms) && searchTerms.length === 2) {
-      if (operator === OperatorType.rangeInclusive) {
-        // example:: (Duration >= 5 and Duration <= 10)
-        query = `(${fieldName} ge ${searchTerms[0]} and ${fieldName} le ${searchTerms[1]})`;
-      } else if (operator === OperatorType.rangeExclusive) {
-        // example:: (Duration > 5 and Duration < 10)
-        query = `(${fieldName} gt ${searchTerms[0]} and ${fieldName} lt ${searchTerms[1]})`;
-      }
+  private getDecimalInputSeparators(columnDef: Column) {
+    const optionName = 'decimalInputSeparators';
```

Comment:
Yes, I was a bit unsure about it as well and I now think I solved this in the wrong place.  It can be better handled in the input control(s) and not in the backend/data access layer.

Perhaps in [inputFilter.ts/handleInputChange](https://github.com/ghiscoding/Angular-Slickgrid/blob/master/src/app/modules/angular-slickgrid/filters/inputFilter.ts#L169) and generalized as `columnDef.filter.searchValueFormatter`, so the code becomes like:

```
...
const enableWhiteSpaceTrim = this.gridOptions.enableFilterTrimWhiteSpace || this.columnFilter.enableTrimWhiteSpace;
    if (typeof value === 'string' && enableWhiteSpaceTrim) {
      value = value.trim();
    }
if (this.columnDef && this.columnDef.filter && typeof this.columnDef.filter.searchValueFormatter === 'function') {
   value = this.columnDef.filter.searchValueFormatter(value);
}
...
```

and in my code base I'll do `filter.searchValueFormatter: (value) => value.replace(/'\,'/g, '.')` on the columns I need this.

What do you think?

Reply:
Moving the logic within the filter would also help in making this available to all backend services (and even local grids), but I wonder if you'll be using this really on a per column basis? wouldn't it be better to have something available in the grid options for all columns to pull the option? in some cases I make it available in both places (prioritizing the column then the grid options). I still wonder though if there's any way we can use the FormatterOptions in the grid options for this enhancement? I know it's not a Formatter but a Filter, but since you're still calling a `searchValueFormatter`... I mean technically speaking when the users formats its data with a comma as decimal, I would assume the same user would want to filter the same way as well, isn't it? I'm of course trying to reuse code whenever I can.

Reply:
It's subtle, but formatter and inputting values is not the same thing here in NL. We typically format/display numbers and currencies as `1.234,56`, but when inputting we don't type in the thousands separator and for the decimal separator both the `,` and the `.` have the same meaning/intent which has to do with that 99% of keyboards sold here are en-US layout which has a `.` on the numeric keypad. So users inputs `1234.56` via the keypad or `1234,56` via the normal keyboard and they expect both to be the same value.

So yes, `searchValueFormatter` is a confusing name. I always struggle with naming things 😜 

But it could very well be an edge case that only exists in NL in financial accounting and the functionality is easy for us to implement by subclassing `filters.filterText`:
```
import { Inject, Injectable, LOCALE_ID } from '@angular/core';
import { Filters } from 'angular-slickgrid';

@Injectable()
export class InputCurrencyFilter extends Filters.inputText {
  constructor(
    @Inject(LOCALE_ID) private _locale: string
) {
    super();
  }

  protected handleInputChange(e: any) {
    const oldValue = e && e.target && e.target.value;

    if (typeof oldValue === 'string' && this._locale.startsWith('nl')) {
       e.target.value = oldValue.replace(/\,/g, '.');;
    }

    super.handleInputChange(e);

    if (e && e.target && e.target.value !== oldValue) {
      e.target.value = oldValue;
    }
  }
}
```  

This is a pretty workable solution and long term I think we'll be doing some more customization in this area. So I'll remove the `getDecimalInputSeparators` from this PR (and fix the coverage as mentioned the other day). 

Thanks for reviewing and pushing back on this PR! And also thank you for an excellent Angular-Slickgrid 🥇

Reply:
Thanks for the kind words, appreciate it 😄 

However the @Injectable won't work because these filters are manually instantiate, I have a Filter Factory but I never found how to make it properly work with DI, so you can see on this [line](https://github.com/ghiscoding/Angular-Slickgrid/blob/master/src/app/modules/angular-slickgrid/filters/filterFactory.ts#L24) that it's newing by hand following specific argument positions without extras, so your new arguments just won't be added at all. I would prefer that we find a solution that works for everyone with some kind of grid options or column options.

Naming is also hard for me, French is my mother tongue so yeah I come up with some weird names too sometime lol.

Reply:
Ah thanks for pointing out that `@Injectable`, yes that doesn't work :) 

This does:
```
column.filter = {
  model: InputCurrencyFilter,
    params: {
      locale: this._locale
     }
};
...            
export class InputCurrencyFilter extends Filters.inputText {
  constructor(_translate: any, _collection: any) {
    super();
  }

  protected handleInputChange(e: any) {
    const locale: string = this.columnFilter.params.locale;
...
```

This is similar to the approach you take here [custom-angularComponentFilter.ts](https://github.com/ghiscoding/Angular-Slickgrid/blob/master/src/app/examples/custom-angularComponentFilter.ts)

Reply:
I can just use `new`
```
model: new InputCurrencyFilter(this._locale)
...
class InputCurrencyFilter {
  constructor(
    private _locale: string
   )
}
```

....or I can do 'transient' 🕺  
 ```
model: this.createInstance(InputCurrencyFilter)

...

private createInstance<T>(token: Type<T>): T {
  return Injector.create({
    providers: [
	{ provide: token, useClass: token }
    ],
    parent: this._injector
  }).get<T>(token, undefined, InjectFlags.Self);
}

@Injectable()
class InputCurrencyFilter {
  constructor(
    @Inject(LOCALE_ID) private _locale: string
   )
}
```

Reply:
Are you extending `Filters.inputText` because `InputFilter` is not Publicly available? I could, probably should, export them publicly if that is not the case already. If I remember correctly, that is why I created 2 indexes in my other lib (Slickgrid-Universal), the regular [index](https://github.com/ghiscoding/slickgrid-universal/blob/master/packages/common/src/filters/index.ts) and [filters.index](https://github.com/ghiscoding/slickgrid-universal/blob/master/packages/common/src/filters/filters.index.ts) so that both ways are available. I probably should add that to my list of reminder of things to do next year ;)

Reply:
Yes exactly, `InputFilter` isn't public.